components/server/src/database/measurements.py

```diff
@@ -17,11 +18,18 @@ def latest_measurement(database: Database, metric: Metric) -> Measurement | None
     return None if latest is None else Measurement(metric, latest)
 
 
-def latest_measurements_by_metric_uuid(database: Database, metric_uuids: list[str]) -> dict[str, Measurement] | None:
+MatchType = dict[str, dict[str, Union[list[str], str]]]
+
+
+def latest_measurements_by_metric_uuid(
+    database: Database, date_time: str, metric_uuids: list[str]
+) -> dict[str, Measurement] | None:
     """Return the latest measurements in a dict with metric_uuids as keys."""
+    metric_uuid_match: MatchType = {"metric_uuid": {"$in": metric_uuids}}
+    date_time_match: MatchType = {"start": {"$lte": date_time}} if date_time else {}
     latest_measurement_ids = database.measurements.aggregate(
         [
-            {"$match": {"metric_uuid": {"$in": metric_uuids}}},
+            {"$match": metric_uuid_match | date_time_match},
```

Comment:
Does this do the same as the dict.update() function?

Reply:
Yup, new Python 3.10 syntax.